src/queue.js

```diff
@@ -100,29 +101,29 @@ class Queue {
       })
       .catch(function(err) {
         if (self.retries > 0) {
-          self.retryCounts[task] = self.retryCounts[task] || 0;
-          if (self.retryCounts[task] < self.retries) {
-            self.retryCounts[task]++;
+          self.retryCounts[cursorIndex] = self.retryCounts[cursorIndex] || 0;
+          if (self.retryCounts[cursorIndex] < self.retries) {
```

Comment:
I made this change so that each task will have a different `retryCount`.
This old behavior caused me trouble because if a path has lots of sub paths to delete, it is expected to have multiple tasks named the same. 
For example, if there are 1000,000 documents under `/a/b`. There should be at least 100 tasks named `/a/b` in the queue, because each `/a/b` will only delete the first 10,000 subtrees.
The old behavior only gave N retries for all 100 tasks, and for sure, the retry allowance is going to run out.

@mbleigh I see you wrote most of `queue.js`. Is hosting depending on this `retryCount` behavior?

Reply:
Hosting definitely makes use of retries, yes. I think this is probably a worthwhile refactor because it will make retries robust for non-string types as well. However, we should definitely make sure this doesn't break Hosting deploys before merging.